build/build.go

```diff
@@ -62,7 +62,14 @@ func NewBuildContext(installSuffix string, buildTags []string) *build.Context {
 // If an error occurs, Import returns a non-nil error and a nil
 // *PackageData.
 func Import(path string, mode build.ImportMode, installSuffix string, buildTags []string) (*PackageData, error) {
-	return importWithSrcDir(path, "", mode, installSuffix, buildTags)
+	wd, err := os.Getwd()
+	if err != nil {
+		// Getwd may fail if we're in GOARCH=js mode. That's okay, handle
+		// it by falling back to empty working directory. It just means
+		// Import will not be able to resolve relative import paths.
+		wd = ""
```

Comment:
@neelance That's exactly what I want to do next, this was just a working intermediate solution. Just checking, but did you consider the commit message of f79744a233766fd8897d5f2dc8f05b6ee0e72d8c before making the comment?

As long as there are no functional issues with the current implementation (I'm not aware of any at this time), I'd prefer to proceed with a smaller scope PR and follow up with breaking the `gopherjs/build` API in future changes, rather than doing everything at once just to resolve #302.

Would it help if I copied the future steps from commit message into a TODO comment?

Or do you feel strongly all that needs to be done before this can be merged?